usaspending_api/database_scripts/job_archive/recompute_all_awards.py

```diff
@@ -309,11 +333,27 @@ def main():
         logging.info("Min ID: {:,}".format(min_id))
         logging.info("Max ID: {:,}".format(max_id))
         logging.info("Total in range: {:,}".format(max_id - min_id + 1))
+        logging.info("Closing time: {}".format(CLOSING_TIME))
 
         batch_min = min_id
+        iteration = 1
         while batch_min <= max_id:
             batch_max = min(batch_min + CHUNK_SIZE - 1, max_id)
-            with Timer("[Awards {:,} - {:,}]".format(batch_min, batch_max), pipe_output=logging.info):
+            if CLOSING_TIME:
+                if ITERATION_ESTIMATED_SECONDS:
+                    curr_time = datetime.now(timezone.utc)
+                    next_run_estimated_end_datetime = curr_time + timedelta(seconds=ITERATION_ESTIMATED_SECONDS)
+                    dt_str = next_run_estimated_end_datetime.isoformat()
+                    if next_run_estimated_end_datetime >= CLOSING_TIME:
+                        logging.info("=> Estimated loop end datetime of: {}".format(dt_str))
+                        logging.info("=> Next IDs to process: {} - {}".format(batch_min, batch_max))
+                        logging.info("=> You don't have to go home, but you can't... stay...... heeeeere")
```

Comment:
```suggestion
                        logging.info("=> Suspending procedure around closing time of {}".format(CLOSING_TIME))
                        logging.info("=> You don't have to go home, but you can't... stay...... heeeeere")
```
I know they would have to add the flag themselves, but future devs (or OPs trying to read logs to audit something) might not appreciate our humor at 2 in the morning why trying to figure out when things stopped running.

Reply:
The irony of your concern is that this log message would **only** be printed if the script stopped _when it was set to stop_

Reply:
but it doesn't explain why. If I'm looking at the last ten runs for some reason, I'm left guessing why it decided to print this and stop all the sudden. I'll be able to control-f the code real quick for this very distinct string, which is why I don't think it's an issue, but we could help an ops guy by being more explicit.